src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,133 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int threshold;
+    private Node<K, V>[] table;
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> node = new Node<K, V>(key, value);
+        V valueByNewKey = getValue(key);
+        if (valueByNewKey != null) {
+            if (Objects.equals(value, valueByNewKey)) {
+                return;
+            }
+            putIntoTable(node, table);
+            return;
+        }
+        if (table == null || size == threshold) {
```

Comment:
table should never be null and you shouldn't check it. Create constructor and initialize it there